sklearn/feature_extraction/tests/test_text.py

```diff
@@ -222,6 +222,27 @@ def test_char_wb_ngram_analyzer():
     assert_equal(cnga(text)[:6], expected)
 
 
+def test_word_ngram_analyzer():
+    cnga = CountVectorizer(analyzer='word', strip_accents='unicode',
+                           ngram_range=(3, 6)).build_analyzer()
+
+    text = "This \n\tis a test, really.\n\n I met Harry yesterday"
+    expected = ['this is test', 'is test really', 'test really met']
+    assert_equal(cnga(text)[:3], expected)
+
+    expected = ['test really met harry yesterday',
+                'this is test really met harry',
+                'is test really met harry yesterday']
+    assert_equal(cnga(text)[-3:], expected)
+
+    cnga = CountVectorizer(input='file', analyzer='word',
+                           ngram_range=(3, 6)).build_analyzer()
+    text = StringIO("A test with a file-like object!")
```

Comment:
I'd rather just use file = StringIO(text) and check that the output with a string and with a file are identical, rather than checking a sample with a different text.